src/coreclr/jit/optimizebools.cpp

```diff
@@ -1438,6 +1478,395 @@ GenTree* OptBoolsDsc::optIsBoolComp(OptTestInfo* pOptTest)
     return opr1;
 }
 
+//-----------------------------------------------------------------------------
+// Reinit:   Procedure that reinitialize IntBoolOpDsc reference
+//
+void IntBoolOpDsc::Reinit()
+{
+    start           = nullptr;
+    end             = nullptr;
+    ctsArrayLength  = 0;
+    lclVarArrLength = 0;
+}
+
+//-----------------------------------------------------------------------------
+// Free:   Procedure that frees IntBoolOpDsc reference
+//
+void IntBoolOpDsc::Free()
```

Comment:
Once you switch the arrays over to using `ArrayStack`, you can get rid of `IntBoolOpDsc::Free` entirely.